castai/resource_workload_scaling_policy.go

```diff
@@ -1058,3 +1170,206 @@ func getWorkloadScalingPolicyByName(ctx context.Context, client sdk.ClientWithRe
 	}
 	return nil, fmt.Errorf("policy with name %q not found", name)
 }
+
+func toAssignmentRules(in map[string]any) (*[]sdk.WorkloadoptimizationV1ScalingPolicyAssignmentRule, error) {
+	if len(in) == 0 {
+		return nil, nil
+	}
+
+	rules := in["rules"].([]any)
```

Comment:
i think this could throw panic

Reply:
the whole pkg is written that way in most places, I think that's shouldn't be an issue as we have a schema that it validated by TF, so here we should get the slice